Foundation/FileManager.swift

```diff
@@ -1381,6 +1381,30 @@ extension FileManager {
         }
         
         override func nextObject() -> Any? {
+
+            func match(filename: String, to options: DirectoryEnumerationOptions, isDir: Bool) -> (Bool, Bool) {
+                var showFile = true
+                var skipDescendants = false
+
+                if isDir {
+                    if options.contains(.skipsSubdirectoryDescendants) {
+                        skipDescendants = true
+                    }
+#if canImport(Darwin)
+                    if options.contains(.skipsPackageDescendants) && filename.hasSuffix(".app") {
```

Comment:
Let's not do this .app stuff at all, even on Darwin. There are a plethora of extensions that can represent a package, and even more non-filename-based conditions. In this instance, I'd say it's better to have consistent behavior between platforms than have an inadequate Darwin behavior.

I know this means that the API's implementation does nothing, but I don't think there's a better alternative at the moment.